### ISSUE TITLE
Unable to mock HandlerMethodArgumentResolver in spring webflux.

### ISSUE THREAD

**Issue by ashishkujoy** [Ninja-Squad/springmockk#40]
While trying to mock `org.springframework.web.reactive.result.method.HandlerMethodArgumentResolver` using @MockkBean getting following error.
```Kotlin
@ActiveProfiles("test")
@WebFluxTest(MyController::class)
@ExtendWith(SpringExtension::class)
@Import(CSRFConfiguration::class)
class SomeTest {
  @MockkBean private lateinit var authenticationTokenDataResolver: HandlerMethodArgumentResolver
}
```

```bash
Caused by: java.lang.IllegalStateException: Failed to load ApplicationContext
at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContext(DefaultCacheAwareContextLoaderDelegate.java:132)
at org.springframework.test.context.support.DefaultTestContext.getApplicationContext(DefaultTestContext.java:123)
at org.springframework.test.context.junit.jupiter.SpringExtension.getApplicationContext(SpringExtension.java:202)
at org.springframework.test.context.junit.jupiter.SpringExtension.resolveParameter(SpringExtension.java:188)
at org.junit.jupiter.engine.execution.ExecutableInvoker.resolveParameter(ExecutableInvoker.java:216)
... 75 more
        Caused by: java.lang.IllegalStateException: Unable to register mock bean org.springframework.web.reactive.result.method.HandlerMethodArgumentResolver expected a single matching bean to replace but found [authenticationPrincipalArgumentResolver, reactiveCurrentSecurityContextArgumentResolver]
at com.ninjasquad.springmockk.MockkPostProcessor.getBeanName(MockkPostProcessor.kt:204)
at com.ninjasquad.springmockk.MockkPostProcessor.registerMock(MockkPostProcessor.kt:158)
at com.ninjasquad.springmockk.MockkPostProcessor.register(MockkPostProcessor.kt:145)
at com.ninjasquad.springmockk.MockkPostProcessor.postProcessBeanFactory(MockkPostProcessor.kt:109)
at com.ninjasquad.springmockk.MockkPostProcessor.postProcessBeanFactory(MockkPostProcessor.kt:94)
at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:286)
at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:174)
at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:706)
at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:532)
at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:747)
at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:397)
at org.springframework.boot.SpringApplication.run(SpringApplication.java:315)
at org.springframework.boot.test.context.SpringBootContextLoader.loadContext(SpringBootContextLoader.java:126)
at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContextInternal(DefaultCacheAwareContextLoaderDelegate.java:99)
at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContext(DefaultCacheAwareContextLoaderDelegate.java:124)
... 79 more
```
I am able to mockk it using 
```Kotlin
@ActiveProfiles("test")
@WebFluxTest(MyController::class)
@ExtendWith(SpringExtension::class)
@Import(CSRFConfiguration::class)
class SomeTest(@Autowired val authenticationTokenDataResolver: HandlerMethodArgumentResolver) {
  @MockkBean private lateinit var authenticationTokenDataResolver: HandlerMethodArgumentResolver
}
 @TestConfiguration
    class ControllerTestConfig {
        @Bean
        fun authenticationTokenDataResolver() = mockk<HandlerMethodArgumentResolver>()
    }
```

**Comment by jnizet** [comment]
Hi @ashishkujoy 

I'm not sure I'll be able to help on that one, especially without a minimal example reproducing the problem..
Here's what I would do (in that order):

1. Wait until the version I just released (2.0.2) is available in Maven central, and then try using it. This release contains bug fixes made in Spring Boot that I reported in SpringMockK. If the issue can't be reproduced anymore, then it means that it was caused by a bug in Spring Boot that has been fixed in Spring Boot, and now been fixed in SpringMockK too.
2. If the issue isn't solved yet, try to reproduce it using Mockito (MockBean instead of MockkBean). If the issue also happens, then the issue is not specific to SpringMockK, but is a Spring Boot issue that you should report there.
3. Create a **minimal** example reproducing the issue, and post it as a github repo. I'll try to reproduce it and see what I can do (no promise!)

Small tip, so that you gain at least something by posting the issue: you don't need the annotation `@ExtendWith(SpringExtension::class)`. The annotation `@WebFluxTest` is a meta-annotation annotated with `@ExtendWith(SpringExtension::class)` already, and thus automatically applies the extension.

**Comment by jnizet** [comment]
Closing due to the absence of feedback
